From ef339bc399d8a69ca474567516cc211af81cbc42 Mon Sep 17 00:00:00 2001
From: Manjunath <manjunathx.b.s@intel.com>
Date: Tue, 4 Dec 2012 17:25:30 +0530
Subject: [PORT FROM MAIN] [PATCH] The wrong format ogg make the music player crash

BZ: 71305

In _vorbis_unpack_books function, when extracting
map backend settings, memory is allocated for the
mapping info structure. But the structure in itself
has pointers to chmuxlist, submaps, coupling_steps
which contains random addresses. So when mapping_info_unpack
returns error, mapping_clear_info is called to free the
memory if the pointer isn't NULL.
The current fix uses calloc instead of malloc to initialize
the mapping info structure to zero.
Signed-off-by: Manjunath <manjunathx.b.s@intel.com>

Change-Id: I3819ad5c80661169d75fb674451414eca63dc308
Orig-Change-Id: I4f0935f79b7e730c2098807aaf6511f7307207ff
Reviewed-on: http://android.intel.com:8080/82058
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 Tremolo/treminfo.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Tremolo/treminfo.c b/Tremolo/treminfo.c
index f7c4973..8474a0b 100644
--- a/Tremolo/treminfo.c
+++ b/Tremolo/treminfo.c
@@ -279,7 +279,7 @@ int _vorbis_unpack_books(vorbis_info *vi,oggpack_buffer *opb){
 
   /* map backend settings */
   ci->maps=oggpack_read(opb,6)+1;
-  ci->map_param=_ogg_malloc(sizeof(*ci->map_param)*ci->maps);
+  ci->map_param=_ogg_calloc(ci->maps,sizeof(*ci->map_param));
   for(i=0;i<ci->maps;i++){
     if(oggpack_read(opb,16)!=0)goto err_out;
     if(mapping_info_unpack(ci->map_param+i,vi,opb))goto err_out;
-- 
1.7.4.1

